### PR TITLE
T/690: Fixed script which validates docs

### LINKED ISSUE #690: Docs valdiation does not break the build
## 🐞 Is this a bug report or feature request? (choose one)

- Bug report

## 💻 Version of CKEditor

Latest master.

## 📋 Steps to reproduce

1. Run `npm run docs:api -- --validate-only`.

## ✅ Expected result

If https://github.com/ckeditor/ckeditor5-utils/issues/209 is not resolved, `npm` should fail.

## ❎ Actual result

`npm` prints the warning but everything is ok.

## 📃 Other details that might be useful

It's a regression after https://github.com/ckeditor/ckeditor5/issues/473.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Script which validates documentation will work properly. Closes #690.

### CODE REVIEW

File: scripts/docs/buildapi.js [hunk 1]
```diff
@@ -24,6 +24,6 @@ module.exports = function buildApiDocs() {
 				'!' + process.cwd() + '/packages/ckeditor5-*/src/lib/**/*.js',
 				'!' + process.cwd() + '/packages/ckeditor5-build-*/src/**/*.js'
 			],
-			validateOnly: process.argv[ 3 ] == '--validate-only'
+			validateOnly: process.argv[ 2 ] == '--validate-only'
```
Comment (ma2ciek) [inline scripts/docs/buildapi.js]: `process.argv.includes( '--validate-only' )` would be safer.

Comment (pomek) [reply scripts/docs/buildapi.js]: Good one!

**Comment by pomek** [comment]
Because some of the docs are invalid, build for this PR should fail:

image

File: scripts/docs/build-api-docs.js [hunk 2]
```diff
@@ -11,4 +11,7 @@
 
 const buildApiDocs = require( './buildapi' );
 
-buildApiDocs();
+buildApiDocs()
+	.catch( () => {
+		process.exitCode = -1;
```
Comment (ma2ciek) [inline scripts/docs/build-api-docs.js]: Not sure about negative error codes. But I guess that everything despite exitCode that is equal to 0 will be treated as an error.

Comment (ma2ciek) [reply scripts/docs/build-api-docs.js]: screen shot 2017-12-11 at 11 49 06

Comment (pomek) [reply scripts/docs/build-api-docs.js]: You're right. We should use `1` error code (http://tldp.org/LDP/abs/html/exitcodes.html).

`1` means _Catchall for general errors_.

**Pull request merged**